spec/acceptance/types/mysql_plugin_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'spec_helper_acceptance'
+
+describe 'mysql_plugin' do
+  describe 'setup' do
+    it 'should work with no errors' do
+      pp = <<-EOS
+        class { 'mysql::server': }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+    end
+  end
+
+  describe 'load plugin' do
+    it 'should work without errors' do
+      pp = <<-EOS
+        mysql_plugin { 'auth_socket':
+          ensure => present,
+          soname => 'auth_socket.so',
+        }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+    end
+
+    it 'should find the plugin' do
+      shell("select plugin_name from information_schema.plugins where plugin_name='auth_socket'") do |r|
```

Comment:
I don't think this would work, since shell() shells out to the sh shell, not the mysql shell. Should look more like https://github.com/puppetlabs/puppetlabs-mysql/blob/master/spec/acceptance/types/mysql_user_spec.rb#L27